src/main/java/io/cdap/plugin/servicenow/source/ServiceNowRecordReader.java

```diff
@@ -48,23 +49,20 @@ public ServiceNowRecordReader(ServiceNowSourceConfig pluginConf) {
 
   @Override
   public void initialize(InputSplit split, TaskAttemptContext context) {
-    this.split = (ServiceNowInputSplit) split;
-    this.pos = 0;
-    restApi = new ServiceNowTableAPIClientImpl(pluginConf.getConnection());
-    tableName = ((ServiceNowInputSplit) split).getTableName();
-    tableNameField = pluginConf.getTableNameField();
-    fetchSchema(restApi);
+    initialize(split);
+    fetchSchema(new ServiceNowJobConfiguration(context.getConfiguration()).getTableInfos(), tableName);
```

Comment:
Little confusing to understand what's going on. fetchSchema takes in a list of tableInfos, gets the schema from the list for given tableName and then initializes the schema field in the RecordReader right?

nit: can get the tableInfos in a separate line and rename the method to fetchAndInitializeSchemas. It would make it a bit more readable.

Reply:
Ack.